controllers/order.js

```diff
@@ -0,0 +1,99 @@
+const asyncHandler = require('express-async-handler');
+const Order        = require('../models/Order');
+
+
+// @route POST /orders/
+// @descp Create a new Order
+// @access Private
+exports.createOrder = asyncHandler(async (req , res , next)=>{
+    const newOrder = new Order(req.body);
+    const savedOrder = await newOrder.save();
+    if(!savedOrder){
+        res.status(500).json({error : "Something went wrong"})
+    }else{
+        res.status(200).json({order : savedOrder})
+    }
+})
+
+// @route GET /orders/:id
+// @descp Get an Order
+// @access Private
+exports.getAnOrder = asyncHandler(async (req , res , next)=>{
+    const orderId = req.params.id;
+    const foundOrder = await Order.findById(orderId);
+    if(!foundOrder){
+        res.status(500).json({error : "something went wrong"});
+    }else{
+        res.status(200).json({order : foundOrder})
+    }
+})
+
+// @route PUT /orders/:id
+// @descp Edit an order
+// @access Private
+exports.updateOrder = asyncHandler(async (req , res , next)=>{
+    const orderId = req.params.id;
+    updatedOrder = await Order.findByIdAndUpdate(orderId , req.body);
```

Comment:
you can check if the `order `exists else return a `404 `error before proceeding next step

Reply:
Do you mean if OrderId is falsy, I return a `404` error?

Reply:
yes I think the case, you receive a false/invalid `orderId ` to update `Order`, doesn't mean internal server error